third_party/terraform/website/docs/index.html.markdown

```diff
@@ -44,6 +40,39 @@ If you have configuration questions, or general questions about using the provid
 * [Terraform's community resources](https://www.terraform.io/docs/extend/community/index.html)
 * [HashiCorp support](https://support.hashicorp.com) for Terraform Enterprise customers
 
+## Releases
+
+Interested in the provider's latest features, or want to make sure you're up to date?
+Check out the [`google` provider changelog](https://github.com/terraform-providers/terraform-provider-google/blob/master/CHANGELOG.md)
+and the [`google-beta` provider changelog](https://github.com/terraform-providers/terraform-provider-google-beta/blob/master/CHANGELOG.md))
+for release notes and additional information.
+
+Per [Terraform Provider Versioning](https://www.hashicorp.com/blog/hashicorp-terraform-provider-versioning),
+the Google provider follows [semantic versioning](https://semver.org/).
+
+In practice, patch / bugfix-only releases of the provider are infrequent. Most
+provider releases are either minor or major releases.
+
+### Minor Releases
+
+The Google provider currently aims to publish a minor release every 2 weeks,
+although the timing of individual releases may differ if required by the
+provider team.
+
+### Major Releases
+
+The Google provider publishes major releases roughly yearly. An upgrade guide
+will be published to help ease you through the transition between the prior
+releases series and the new major release.
+
+During major releases, all current deprecation warnings will be resolved,
```

Comment:
If we could soften this a bit so we're not _promising_ this, that'd be awesome. "You should expect all current deprecation warnings to be resolved" or "all current deprecation warnings will likely be resolved", etc.

Reply:
We're going to make a pass over every deprecation message regardless, and I see one of the resolution paths as "maintain the current deprecation message".

Merging this based on that being an option- if you feel that we're misrepresenting what we'll do in practice, feel free to PR a fix.